kale/loaddata/multi_domain.py

```diff
@@ -229,3 +232,249 @@ def _split_dataset_few_shot(dataset, n_fewshot, random_state=None):
+            sub_domain_set (list): A list of domain names, which should be a subset of domains in image folders.
+                Defaults to None.
+            sub_class_set (list): A list of class names, which should be a subset of classes in image folders.
+                Defaults to None.
+            is_valid_file (callable, optional): A function that takes path of a file and check if the file is a valid
+                file (used to check of corrupt files) both extensions and is_valid_file should not be passed.
+         Attributes:
+            classes (list): List of the class names sorted alphabetically.
+            class_to_idx (dict): Dict with items (class_name, class_index).
+            samples (list): List of (sample path, class_index) tuples
+            targets (list): The class_index value for each image in the dataset
+            domains (list): List of the domain names sorted alphabetically.
+            domain_to_idx (dict): Dict with items (domain_name, domain_index).
+            domain_labels (list): The domain_index value for each image in the dataset
+        """
+
+    def __init__(
+        self,
+        root: str,
+        loader: Callable[[str], Any] = default_loader,
+        extensions: Optional[Tuple[str, ...]] = IMG_EXTENSIONS,
+        transform: Optional[Callable] = None,
+        target_transform: Optional[Callable] = None,
+        sub_domain_set=None,
+        sub_class_set=None,
+        is_valid_file: Optional[Callable[[str], bool]] = None,
+        return_domain_label: Optional[bool] = False,
+        split_train_test: Optional[bool] = False,
+        split_ratio: Optional[float] = 0.8,
+    ) -> None:
+        super(MultiDomainImageFolder, self).__init__(root, transform=transform, target_transform=target_transform)
+        domains, domain_to_idx = self._find_classes(self.root)
+        if type(sub_domain_set) == list:
+            for domain_name in sub_domain_set:
+                if domain_name not in domains:
+                    raise ValueError("Domain %s not in the image directory" % domain_name)
+            domains = sub_domain_set
+            domain_to_idx = {domain_name: i for i, domain_name in enumerate(sub_domain_set)}
+
+        classes, class_to_idx = self._find_classes(os.path.join(self.root, domains[0]))
+        if type(sub_class_set) == list:
+            for class_name in sub_class_set:
+                if class_name not in classes:
+                    raise ValueError("Class %s not in the image directory" % class_name)
+            classes = sub_class_set
+            class_to_idx = {class_name: i for i, class_name in enumerate(sub_class_set)}
+        # for domain in domains:
+        #     domain_path = os.path.join(self.root, domain)
+        #     classes_, class_to_idx_ = self._find_classes(domain_path)
+        #     if not classes == classes_:
+        #         raise ValueError("Classes for different domains are expected to be the same.")
```

Comment:
What the above commented lines for? Add some explanation to help readers/users understand.